go/exocom/src/client_registry/client_registry.go

```diff
@@ -0,0 +1,77 @@
+package clientRegistry
+
+import "encoding/json"
+
+// Client is the combination of a client name, service type and internal namespace
+type Client struct {
```

Comment:
Thoughts around extracting each struct into its own file with the struct name as the file name? That's how it is done in Ruby, and it's very helpful in finding implementations.